### DISCUSSION TITLE
Project cleanup?

### DISCUSSION THREAD

**Discussion by jmsche** [spatie/dropbox-api#122] [Q&A] [answered]
Hi,

I was wondering if some "cleanup" could be done on the project?

I was thinking of the following:

* allow PHP ^8.1 (so remove support for PHP 7.1+ & PHP 8.0) as they are not maintained anymore (PHP 8.0 is currently under security support: https://www.php.net/supported-versions.php)
* allow the php-cs-fixer workflow to run during pull requests
* drop windows from tests workflow, not sure why this was added in the first place?
* add types everywhere it is possible
* probably try to upgrade to PHPUnit 10

I'd happily volunteer to implement these steps.

**Comment by freekmurze** [comment, marked as answer]
I'm always up for modernising packages 🙂 In addition to this, lets move to Laravel Pint and Pest.

Feel free to send a PR for all of this! 👍

**Reply by jmsche** [reply]
Haven't used these yet as I primarily develop using Symfony, but I'll give it a try. :)

**Reply by freekmurze** [reply]
I bet you'll love Pest 🙂 

You can find the Pint workflow here: https://github.com/spatie/laravel-prometheus/blob/main/.github/workflows/fix-php-code-style-issues.yml

**Reply by jmsche** [reply]
Would you prefer several PRs (like I just did), or one PR but with meaningful commits? (so not a gigantic one :smile: )

**Reply by freekmurze** [reply]
For the modernisation of this package, let's just do 1 PR. Do not make any functional changes to the package itself, those can go in a follow PR if needed.

**Reply by jmsche** [reply]
Didn't plan to make functional changes :) 

Thanks for the quick feedback!
